### PR TITLE
Fix(BB-651): Error handling for not found editor

### PR DESCRIPTION
**Pull request by tr1ten**
### Problem

Since function execution doesn't stop after next(err) thus causing issues

Relevant ticket BB-651: 502 when trying to load an editor by name

### Areas of Impact

editor route

### CODE REVIEW

File: src/server/routes/editor.js [hunk 1]
```diff
@@ -257,10 +257,12 @@ function achievementColToEditorGetJSON(achievementCol) {
...
 }
 
-router.get('/:id', async (req, res, next) => {
+router.get('/:id', (req, res, next) => {
 	const {AchievementUnlock, Revision} = req.app.locals.orm;
 	const userId = parseInt(req.params.id, 10);
-
+	if (!userId) {
+		return next(new error.BadRequestError('Editor Id is not valid!', req));
+	}
```
Comment (MonkeyDo) [inline src/server/routes/editor.js:265]: Yes, let's keep this !

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
We're slowly moving the codebase to use async/await wherever we can, so I don't want us to revert to using promise chains.

I believe what is missing is a `try {…} catch(err) {next(err)}`  block around the meat of the query, wrapping around the lines from `const editorJSONPromise`  to `res.send(target({…}));`

Finally in this route we should remove the instances of `.catch(next)`; since we wrap the whole thing, we want errors to bubble up to our main try-catch block and call `next` in the catch part.

Do let me know if I wasn't clear :)

**Comment by tr1ten** [comment]
I wonder, in general what would be more efficient- awaiting on each promise or awaiting on promise.all, any difference performance wise?

**Review by MonkeyDo — APPROVED** [review]
Thank you !
That fixes that issue !

**Pull request merged**